jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1270,7 +1270,8 @@ protected KeyStore loadTrustStore(Resource resource) throws Exception
     @Deprecated
     protected X509ExtendedKeyManager newSniX509ExtendedKeyManager(X509ExtendedKeyManager keyManager)
     {
-        throw new UnsupportedOperationException("X509ExtendedKeyManager only supported on Server");
+        LOG.warn("Using Deprecated SslContextFactory implementation, SNI does not work in this context, use SslContextFactory.Server instead.");
+        return null;
```

Comment:
Looking at https://github.com/eclipse/jetty.project/blob/8a305438454c36fb0fc0db3447227a75c79ab838/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java#L1255, would it make sense to rather return the supplied `keyManager` and thereby degrade more gracefully?

Reply:
returning null is just going from UnsupportedOperationException to NullPointerException in affected calling code. I agree a more graceful behaviour seems appropriate.

Reply:
We can either return null, or throw an Exception.
Returning the unwrapped `X509ExtendedKeyManager` means broken behaviors.